Src/AutoNSubstituteUnitTest/Scenario.cs

```diff
@@ -0,0 +1,97 @@
+﻿using NSubstitute;
+using NSubstitute.Exceptions;
+using Ploeh.AutoFixture.Xunit;
+using Xunit;
+using Xunit.Extensions;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class Scenario
+    {
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void InterfacesAreSubstitutedByDefault(IInterface i, object obj)
+        {
+            i.Method(obj);
+            i.Received().Method(obj);
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void AbstractTypesAreSubstitutedByDefault(AbstractType t, object obj)
+        {
+            t.AbstractMethod(obj);
+            t.Received().AbstractMethod(obj);
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void ConcreteTypesAreNotSubstitutedByDefault(ConcreteType t, object obj)
+        {
+            t.VirtualMethod(obj);
+            Assert.Throws<NotASubstituteException>(() => t.Received().VirtualMethod(obj));
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void ConcreteTypesAreSubstitutedWhenParameterIsMarkedWithSubstituteAttribute(
+            [Substitute]ConcreteType t, object obj)

```

Comment:
As someone who chose NSubstitute for its terse, low-ceremony API, I'd find having to use `Substitute<T>` awkward. The [. Operator](https://msdn.microsoft.com/en-us/library/6zhxzbds(v=vs.71).aspx) cannot be overloaded to hide the wrapper type, so the `Substitute<T>` would bleed into my test code. It would start looking Moq-ish, only more confusing, because `Received()` has to be called for the actual substitute instance of type `T` instead of the instance of `Substitute<T>`.

As we discussed in #420, the `[Frozen]` attribute cannot be combined with `[Substitute]` or other creation attributes today because `FreezingCustomization` uses a fixture customization instead of fixture behavior. 

While the type-based approach would be sufficient to support the `[Frozen]`/Substitute scenario today, the attribute-based approach is more open to future extension. The `SubstituteAttribute` attribute can be parameterized or combined other attributes to support creation of [partial substitutes](http://nsubstitute.github.io/help/partial-subs/), [unconfigured substitutes](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoNSubstitute/AutoConfiguredNSubstituteCustomization.cs) and, once #420 is addressed,  _frozen unconfigured partial substitutes_ for those who need them.